### ISSUE TITLE
mockReturnValue is not typesafe

### ISSUE THREAD

**Issue by peterreisz** [hirezio/auto-spies#51]
This compiles, but it shouldn't:

```ts
import { createSpyFromClass } from 'jest-auto-spies';

class Foo {
  bar(param: string): string {
    return '...';
  }
}

const foo = createSpyFromClass(Foo);
foo.bar.mockReturnValue(1);
foo.bar.calledWith(1).mockReturnValue(1);
```

When there is a strong type system we shouldn't write tests for invalid cases.

typescript version: 4.2.4
jest-auto-spies version: 1.6.5

**Comment by meksof** [comment]
Hi Shai and Guillermo,
I am interested to take over this issue.

Sofiane

Le mer. 11 août 2021 à 14:41, peterreisz ***@***.***> a
écrit :

**Comment by shairez** [comment]
Thanks @peterreisz for the report

And thanks a lot @meksof for the help!
I've assigned this issue to you
Please let us know if you need anything

Thanks again! ❤️ 💪

**Comment by azaeng04** [comment]
Has this been resolved? Just noticed this today.

**Comment by mopinsk** [comment]
I think I resolved it in my pull request #73 

Perhaps @shairez could give some feedback. 😉

**Comment by shairez** [comment]
Thanks for the PR @mopinsk ! 🙏
